src/robotFactory.js

```diff
@@ -1,13 +1,74 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0, y: coords.y || 0,
+    };
```

Comment:
Too complex to write in one line.
```suggestion
    this.coords = {
      x: coords.x || 0,
      y: coords.y || 0,
    };
```